### DISCUSSION TITLE
window-padding-balance putting all padding in the bottom of the window

### DISCUSSION THREAD

**Discussion by Joao-Queiroga** [ghostty-org/ghostty#3941] [Q&A]
The window-padding-balance option is puttting all the y padding in the bottom of the window.
I tried this with some high values(20) and the top is allways on edge and the bottom has the padding instead of being centralized. 

![2024-12-29T17:46:41,902263853-03:00](https://github.com/user-attachments/assets/a75b7b83-e519-42fb-9c48-8bce36078373)

**Comment by CRAG666** [comment]
Same error

**Comment by ormai** [comment]
The horizontal padding gets distributed uniformly, while the vertical padding is not.

Here's the grid when I select all the cells

![Screenshot From 2025-02-16 20-03-09](https://github.com/user-attachments/assets/73b420c9-0f4b-46e4-a231-7677822ed9bd)

**Comment by Garmelon** [comment]
From experimentation, the current top padding seems to be the minimum of the horizontal padding and the actual top padding. In other words, `top-padding = min(leftover_horizontal_space/2, leftover_vertical_space/2)` while `left_padding = leftover_horizontal_space/2`.

The source code seems to confirm this, and also that it is intentional:
https://github.com/ghostty-org/ghostty/blob/2f63f840de09bd67998e4dafb2666e7057b60cc0/src/renderer/size.zig#L265-L270

This doesn't quite match the wording "to minimize imbalance on one side" in the description of the config option:
```toml
# The viewport dimensions are usually not perfectly divisible by the cell
# size. In this case, some extra padding on the end of a column and the bottom
# of the final row may exist. If this is `true`, then this extra padding
# is automatically balanced between all four edges to minimize imbalance on
# one side. If this is `false`, the top left grid cell will always hug the
# edge with zero padding other than what may be specified with the other
# `window-padding` options.
```

For example, in my current setup, there's very little horizontal padding but almost a full row of padding at the bottom. In the extreme case of no horizontal padding, the `window-padding-balance` has no effect at all, which seems surprising and wrong.

I personally would prefer if vertical and horizontal padding were balanced independently of another.

**Reply by Cosmic-Empath** [reply]
Window padding wasn't even for me until I added a zero to the default x & y values. This worked for me:

theme = wallbash-gtk
font-size = 8.5
background-blur-radius = 20
mouse-hide-while-typing = true
window-decoration = true
background-opacity = 0.8
window-padding-x = 20
window-padding-y = 20
window-padding-balance = true

**Reply by KorigamiK** [reply]
this is the case for me as well. This is not how i would expect the padding balance to work.

**Reply by WieeRd** [reply]
If it's intentional I wish it provided another option to customize that behavior as well.
Something like `window-padding-balance-prefer-bottom = false`.

**Comment by midgramr** [comment]
Still having this issue. Any fixes so far?
